stetho/src/main/java/com/facebook/stetho/inspector/elements/android/ViewDescriptor.java

```diff
@@ -80,4 +327,60 @@ private static String capitalize(String str) {
     buffer.setCharAt(0, Character.toTitleCase(buffer.charAt(0)));
     return buffer.toString();
   }
+
+  private class FieldBackedCSSProperty extends ViewCSSProperty {
+    private final Field mField;
+
+    public FieldBackedCSSProperty(
+        Field field,
+        String cssName,
+        @Nullable ViewDebug.ExportedProperty annotation) {
+      super(cssName, annotation);
+      mField = field;
+      mField.setAccessible(true);
+    }
+
+    @Override
+    public Object getValue(View view) throws InvocationTargetException, IllegalAccessException {
+      return mField.get(view);
+    }
+  }
+
+  private class MethodBackedCSSProperty extends ViewCSSProperty {
+    private final Method mMethod;
+
+    public MethodBackedCSSProperty(
+        Method method,
+        String cssName,
+        @Nullable ViewDebug.ExportedProperty annotation) {
+      super(cssName, annotation);
+      mMethod = method;
+      mMethod.setAccessible(true);
+    }
+
+    @Override
+    public Object getValue(View view) throws InvocationTargetException, IllegalAccessException {
+      return mMethod.invoke(view);
+    }
+  }
+
+  private abstract class ViewCSSProperty {
+    private final String mCSSName;
+    private final ViewDebug.ExportedProperty mAnnotation;
+
+    public ViewCSSProperty(String cssName, @Nullable ViewDebug.ExportedProperty annotation) {
+      mCSSName = cssName;
+      mAnnotation = annotation;
+    }
+
+    public String getCSSName() {
```

Comment:
we should make this final, since our specific intention is to *not* allow derived classes to override this behavior